x-pack/plugins/canvas/canvas_plugin_src/uis/arguments/select.js

```diff
@@ -13,8 +13,15 @@ import { ArgumentStrings } from '../../../i18n';
 
 const { Select: strings } = ArgumentStrings;
 
-const SelectArgInput = ({ typeInstance, onValueChange, argValue, argId }) => {
-  const choices = typeInstance.options.choices.map(({ value, name }) => ({ value, text: name }));
+const SelectArgInput = ({
+  typeInstance,
+  onValueChange,
+  argValue,
+  argId,
+  choises: dynamicChoises,
+}) => {
+  const existChoises = typeInstance.options.choices ?? dynamicChoises?.[typeInstance.name] ?? [];
```

Comment:
Can you explain where the dynamic choices option comes from? I'm not sure I'm following why it's needed.

Reply:
Oh, it seems to me that is from the version with expression builder. I'll remove it. Before, it was allowing to pass some options to `select` from the `resolve` method of the model.

Reply:
Completely removed the current part of the code.